packages/app-store/ee/routing-forms/playwright/tests/basic.e2e.ts

```diff
@@ -106,6 +107,59 @@ test.describe("Routing Forms", () => {
       await page.isVisible("text=ERROR 404");
     });
 
+    // TODO: Move these tests to @calcom/app-store/typeform
+    test.describe("Typeform App", () => {
+      test("should copy link in editing area", async ({ page, context }) => {
+        await page.waitForSelector('[data-testid="empty-screen"]');
+        context.grantPermissions(["clipboard-read", "clipboard-write"]);
+        await page.goto(`/apps/typeform`);
+        await page.click('[data-testid="install-app-button"]');
+        await page.waitForNavigation({
+          url: (url) => url.pathname === `/apps/typeform/how-to-use`,
+        });
+
+        await page.goto(`/apps/routing-forms/forms`);
+        const formId = await addForm(page);
+        await fillForm(page, {
+          description: "",
+          field: { label: "test", typeIndex: 1 },
+        });
+
+        await page.click('button[tooltip="Delete"] ~ button[data-state="closed"]');
+        await page.locator("text=Copy Typeform Redirect URL").click();
+        const text = await page.evaluate(async () => {
+          return navigator.clipboard.readText();
+        });
+        expect(text).toBe(`${CAL_URL}/router?form=${formId}&test={Recalled_Response_For_This_Field}`);
+      });
+
+      test("should copy link in RoutingForms list", async ({ page, context }) => {
+        await page.waitForSelector('[data-testid="empty-screen"]');
+        context.grantPermissions(["clipboard-read", "clipboard-write"]);
+
+        await page.goto(`/apps/typeform`);
```

Comment:
typeform app installation can be abstracted out in a fn like `installTypeformApp` that can be reused